app/utils/checkOrigin.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const {parse} = require('url');
+
+const {default: logger} = require('@financial-times/n-logger');
+
+const defaultMethods = [
+    'GET',
+    'POST',
+    'PUT',
+    'DELETE'
+];
+
+const defaultDomain = 'ft.com';
+
+module.exports = ({
+    methods = defaultMethods,
+    domain = defaultDomain
+} = {}) => (request, response, next) => {
+
+    const checkDomain = url =>
+        Boolean(url && parse(url).hostname.endsWith(`.${domain}`));
+
+    if (response.locals && response.locals.flags && response.locals.flags.disableSudsOriginCheck) {
+        return next();
+    }
+
+    const {
+        originalUrl,
+        method
+    } = request;
+
+    if (!methods.includes(method)) {
+        return next();
+    }
+
+    const origin = request.get('origin');
+    const referer = request.get('referer');
+
+    const originIsGood = checkDomain(origin);
+    const refererIsGood = checkDomain(referer);
+
+    if (!origin) {
```

Comment:
In all the 4 cases we always logs  originalUrl, method, referer ... that could be optimised a bit with something like:

```
const requestCommonPayload = {
           originalUrl,
            method,
            referer
};
```
and the spread it while logging in the 4 cases:
```
logger.warn('Request does not have an origin.', {
            event: 'NO_REQUEST_ORIGIN',
            ...requestCommonPayload
}
```
This wil simplify readability of the loggers bit for what it's most important.

Reply:
IMHO, DRY doesn't always imply readability, as it requires cross referencing around the code, but in this case I don't think it matters either way.